### DISCUSSION TITLE
[BUG] Environment Variables Not Loading with MikroORM CLI v6.0.5

### DISCUSSION THREAD

**Discussion by tufailra97** [mikro-orm/mikro-orm#5178] [General]
### Describe the bug

I recently updated the MikroORM dependencies in my NestJS project from version `5.7.12` to `6.0.5`. Post-update, I am encountering an issue with the MikroORM CLI (v6.0.5) where it fails to load environment variables. This problem is specifically observed when running MikroORM CLI commands for migrations.

**Error Encountered:**
```ts
Error: No database specified, please fill in `dbName` or `clientUrl` option
```

In the previous setup with MikroORM `v5.7.12`, everything worked seamlessly. Environment variables were properly loaded and utilized by the MikroORM configuration. The issue surfaced only after upgrading to MikroORM `v6.0.5`, specifically when executing migration commands via the MikroORM CLI.

Here's a brief overview of the changes:

**Working Configuration (`v5.7.12`):**
- MikroORM packages (core, entity-generator, migrations, etc.) at version `5.7.12`
- MikroORM CLI version: `5.7.12`
- NestJS version: `10.0.0`

**Non-Working Configuration (`v6.0.5`):**
- MikroORM packages (core, entity-generator, migrations, etc.) upgraded to `6.0.5`
- MikroORM CLI version: `6.0.5`
- NestJS version: `10.3.1`
- Configuration file unchanged

The environment variables are correctly set and function as expected in other parts of the application, but they fail to load when using MikroORM CLI commands in version `6.0.5`. This suggests a potential issue with environment variable handling or a compatibility problem in the latest version of MikroORM CLI.

Could this be a bug? or might be missing something?

### Reproduction

**_mikro orm config:_** 
```ts
// dir: src/mikro-orm.config.ts

import { LoadStrategy } from '@mikro-orm/core';
import { EntityGenerator } from '@mikro-orm/entity-generator';
import { Migrator } from '@mikro-orm/migrations';
import { defineConfig } from '@mikro-orm/postgresql';
import { TsMorphMetadataProvider } from '@mikro-orm/reflection';
import { SeedManager } from '@mikro-orm/seeder';
import { SqlHighlighter } from '@mikro-orm/sql-highlighter';

const migrationRegex = /^[a-zA-Z_]*$/;

export default defineConfig({
  clientUrl: process.env.DATABASE_URL,
  entities: ['dist/**/*.entity.js'],
  entitiesTs: ['src/**/*.entity.ts'],
  debug: process.env.DEBUG_DB_QUERY === 'true' ? true : false,
  loadStrategy: LoadStrategy.JOINED,
  baseDir: process.cwd(),
  highlighter: new SqlHighlighter(),
  metadataProvider: TsMorphMetadataProvider,
  // @ts-expect-error nestjs adapter option
  registerRequestContext: false,
  migrations: {
    snapshot: false,
    path: './src/migrations',
    tableName: 'migrations',
    fileName: (timestamp: string, name?: string) => {
      if (!name) {
        throw new Error(
          'You must specify the migration name! Example: mikro-orm migration:create --name=...'
        );
      }

      if (!migrationRegex.test(name)) {
        throw new Error(
          'Name can only contain letters! Example: "added_user_table"`'
        );
      }

      const migrationName = `Migration_${timestamp}_${name}`;

      return migrationName;
    }
  },
  extensions: [Migrator, EntityGenerator, SeedManager]
});
```

**_package.json_**
```json
{
  "scripts": {
    "db:migrate:create": "npx mikro-orm migration:create",
    "db:migrate:check": "npx mikro-orm migration:check",
    "db:migrate:up": "npx mikro-orm migration:up"
  },
  "dependencies": {
    "@fastify/cookie": "9.3.1",
    "@fastify/static": "6.12.0",
    "@mikro-orm/core": "6.0.5",
    "@mikro-orm/entity-generator": "6.0.5",
    "@mikro-orm/migrations": "6.0.5",
    "@mikro-orm/nestjs": "5.2.3",
    "@mikro-orm/postgresql": "6.0.5",
    "@mikro-orm/reflection": "6.0.5",
    "@mikro-orm/seeder": "6.0.5",
    "@mikro-orm/sql-highlighter": "1.0.1",
    "@nestjs/common": "10.3.1",
    "@nestjs/config": "3.1.1",
    "@nestjs/core": "10.3.1",
    "@nestjs/jwt": "10.2.0",
    "@nestjs/passport": "10.0.3",
    "@nestjs/platform-fastify": "10.3.1",
    "reflect-metadata": "0.2.1",
    "rxjs": "7.8.1"
  },
  "devDependencies": {
    "@mikro-orm/cli": "6.0.5",
    "@nestjs/cli": "10.3.0",
    "@nestjs/schematics": "10.1.0",
    "@nestjs/testing": "10.3.1",
    "@swc/cli": "0.3.4",
    "@swc/core": "1.3.106",
    "@types/node": "20.11.9",
    "constructs": "10.3.0",
    "source-map-support": "0.5.21",
    "ts-loader": "9.5.1",
    "ts-node": "^10.9.2",
    "tsconfig-paths": "4.2.0",
    "typescript": "5.3.3"
  },
  "mikro-orm": {
    "useTsNode": true,
    "tsConfigPath": "./tsconfig.json",
    "configPaths": ["src/mikro-orm.config.ts", "dist/mikro-orm.config.js"]
  }
}
```

**_tsconfig.json_**
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false,
    "paths": {
      "src/*": ["src/*"]
    }
  }
}
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.0.5

### Node.js version

20

### Operating system

Mac OS 14.1.2

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You haven't mentioned it, but my guess is you have a `.env` file, right? Those are no longer loaded, it's your responsibility. The ORM will only read the `MIKRO_ORM_` prefixed env vars out of it automatically, not your custom env vars.

**Reply by tufailra97** [reply]
Yes, that's correct. I am currently loading the environment variables from a `.env` file. May I know the reason behind this change?

**Reply by B4nan** [reply]
It was always just a side effect, and it wasn't possible to disable behavior before. It's not the ORM's responsibility to load your `.env` files into the context. You should load your `.env` files explicitly, if you want this behavior, just add a single line into your ORM config that does it explicitly.

**Reply by tufailra97** [reply]
I undestand. I just wanted to know the reason behind this, as I've noticed a few projects have taken this approach.

**Reply by JordanPlayz158** [reply]
It would also be nice to have this in the v6 upgrade guide (unless I missed it) https://mikro-orm.io/docs/upgrading-v5-to-v6 as this issue took quite a bit to look into to find this bug/discussion.

**Reply by B4nan** [reply]
Here you go b6809f92e99114964dcebd38a5388bb65bc2bfa1

**Reply by JordanPlayz158** [reply]
Thank you! <3
